ts/features/messages/screens/MessagesHomeScreen.tsx

```diff
@@ -1,17 +1,102 @@
+import { MESSAGES_ROUTES } from "../navigation/routes";
+import { resetMessageArchivingAction } from "../store/actions/archiving";
+import {
+  isArchivingInProcessingModeSelector,
+  isArchivingInSchedulingModeSelector
+} from "../store/reducers/archiving";
 
 export const MessagesHomeScreen = () => {
+  const store = useIOStore();
+  const dispatch = useIODispatch();
+  const navigation = useIONavigation();
+
   const pagerViewRef = useRef<PagerView>(null);
   useEngagementScreen();
+
+  /* CODE RELATED TO THE HEADER -- START */
+
+  const currentRoute = MESSAGES_ROUTES.MESSAGES_HOME;
+  const { actionHelp, actionSettings, alertProps } =
+    useHeaderFirstLevelProps(currentRoute);
+  const navigateToSettingMainScreen = useNavigateToSettingMainScreen();
+
+  const canNavigateIfIsArchivingCallback = useCallback(() => {
+    const state = store.getState();
+    // If the system is busy archiving or restoring messages,
+    // a new action (e.g. searching or navigating to settings) cannot be initiated
+    const isProcessingArchiveQueue = isArchivingInProcessingModeSelector(state);
+    if (isProcessingArchiveQueue) {
+      return false;
+    }
+    // If the user was choosing which messages to archive/restore,
+    // disable it before starting a new action (e.g. searching or
+    // navigating to settings), as the tab bar at the bottom is
+    // hidden and the new action could trigger a navigation flow back
+    // to another tab on the main screen without this bar being displayed.
+    const isSchedulingArchiving = isArchivingInSchedulingModeSelector(state);
+    if (isSchedulingArchiving) {
+      // Auto-reset does not provide feedback to the user
+      dispatch(resetMessageArchivingAction(undefined));
+    }
+    return true;
+  }, [dispatch, store]);
+
+  const messageSearchCallback = useCallback(() => {
+    if (canNavigateIfIsArchivingCallback()) {
+      navigation.navigate(MESSAGES_ROUTES.MESSAGES_SEARCH);
+    }
+  }, [canNavigateIfIsArchivingCallback, navigation]);
+
+  const navigateToSettingMainScreenFromMessageSection = useCallback(() => {
+    if (canNavigateIfIsArchivingCallback()) {
+      navigateToSettingMainScreen();
+    }
+  }, [canNavigateIfIsArchivingCallback, navigateToSettingMainScreen]);
+
+  const settingsActionInMessageSection: HeaderActionProps = useMemo(
+    () => ({
+      ...actionSettings,
+      onPress: navigateToSettingMainScreenFromMessageSection
+    }),
+    [navigateToSettingMainScreenFromMessageSection, actionSettings]
+  );
+
+  const searchMessageAction: HeaderActionProps = useMemo(
+    () => ({
+      icon: "search",
+      accessibilityLabel: I18n.t("global.accessibility.search"),
+      onPress: messageSearchCallback
+    }),
+    [messageSearchCallback]
+  );
+
+  useHeaderFirstLevel({
+    title: I18n.t("messages.contentTitle"),
+    ignoreSafeAreaMargin: !!alertProps,
+    type: "threeActions",
+    firstAction: actionHelp,
```

Comment:
I guess we need to define the `ignoreSafeAreaMargin` attribute here

Reply:
See the comment above (but in this case I completely missed the prop, originally present)